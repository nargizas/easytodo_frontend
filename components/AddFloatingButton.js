import { View, Text, Pressable, StyleSheet, Image } from 'react-native';

function AddFloatingButton({ onPress }) {
    return (
        <View style={styles.buttonOuterContainer}>
            <Pressable style={({ pressed }) => pressed ? [styles.buttonInnerContainer, styles.pressed] : styles.buttonInnerContainer} onPress={onPress} android_ripple={{ color: '#ca9200' }}>
                <Image source={require('../assets/images/plus.png')} resizeMode='center' style={styles.image} />
            </Pressable>
        </View>
    );
}

export default AddFloatingButton;

const styles = StyleSheet.create({
    buttonOuterContainer: {
        borderRadius: 36,
        overflow: 'hidden',
    },
    buttonInnerContainer: {
        backgroundColor: '#FFB800',
        paddingVertical: 16,
        paddingHorizontal: 16,
        elevation: 2,
    },
    image: {
        width: 32,
        height: 32
    },
    pressed: {
        opacity: 0.75
    }
})