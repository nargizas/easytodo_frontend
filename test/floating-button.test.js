import React from 'react';
import renderer from 'react-test-renderer';
import AddFloatingButton from '../components/AddFloatingButton';

it('renders correctly', () => {
  const tree = renderer.create(<AddFloatingButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
