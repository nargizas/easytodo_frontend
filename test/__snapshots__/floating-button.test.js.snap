// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "borderRadius": 36,
      "overflow": "hidden",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#FFB800",
        "elevation": 2,
        "paddingHorizontal": 16,
        "paddingVertical": 16,
      }
    }
  >
    <Image
      resizeMode="center"
      source={1}
      style={
        {
          "height": 32,
          "width": 32,
        }
      }
    />
  </View>
</View>
`;
